client/packages/system/src/Stock/api/hooks/useStockLine.ts

```diff
@@ -0,0 +1,92 @@
+import { useCallback, useState } from 'react';
+import {
+  FnUtils,
+  setNullableInput,
+  useMutation,
+} from '@openmsupply-client/common';
+import {
+  InventoryAdjustmentReasonRowFragment,
+  StockLineRowFragment,
+} from '../../..';
+import { STOCK_LINE } from './keys';
+import { useStockGraphQL } from '../useStockGraphQL';
+
+export interface DraftStockLine extends StockLineRowFragment {
+  inventoryAdjustmentReason: InventoryAdjustmentReasonRowFragment | null;
+}
+
+export function useStockLine() {
+  const [draft, setDraft] = useState<DraftStockLine>({
+    __typename: 'StockLineNode',
+    id: '',
+    itemId: '',
+    onHold: false,
+    packSize: 0,
+    sellPricePerPack: 0,
+    costPricePerPack: 0,
+    totalNumberOfPacks: 0,
+    availableNumberOfPacks: 0,
+    storeId: '',
+    item: {
+      __typename: 'ItemNode',
+      code: '',
+      name: '',
+    },
+    inventoryAdjustmentReason: null,
+  });
+
+  const { mutateAsync: createMutation } = useCreate();
+
+  const create = useCallback(async () => {
+    await createMutation(draft);
+  }, [draft, createMutation]);
+
+  return {
+    draft,
+    setDraft,
+    create,
+  };
+}
+
+const useCreate = () => {
```

Comment:
I'd be inclined to just keep this inside the main hook above -- any particular reason to split it into its own hook? Just personal preference?

(It would save having to use `useCallback` too, I think -- I'm, assuming `useMutation` has some memo-isation mechanisms internally)

Reply:
i was making a similar comment earlier, and tried to change as example, it ended up looking slightly worse. I think it was mainly due to storeId, and then onSuccess.

Reply:
Main motivation was that if the main hook starts exposing more methods, the main API is easy to see at the top of the file. Though probably don't really need the useCallback now you mention it...